src/libexpr/primops.cc

```diff
@@ -4074,37 +4074,151 @@ void EvalState::createBaseEnv()
 
     /* `builtins' must be first! */
     v.mkAttrs(buildBindings(128).finish());
-    addConstant("builtins", v);
+    addConstant("builtins", v, {
+        .type = "attribute set",
+        .doc = R"(
+          Contains all the [built-in functions](./builtins.md) and values, in order to avoid polluting the global scope.
+
+          Since built-in functions were added over time, [testing for attributes](./operators.md#has-attribute) in `builtins` can be used for graceful fallback on older Nix installations:
+
+          ```nix
+          if builtins ? getEnv then builtins.getEnv "PATH" else ""
+          ```
+        )",
+    });
 
     v.mkBool(true);
-    addConstant("true", v);
+    addConstant("true", v, {
+        .type = "boolean",
+        .doc = R"(
+          Primitive value.
+
+          It can be returned by [comparison operators](...) and used in [conditional expressions](...).
```

Comment:
That was not meant as a literal suggestion. Please fix the links and apply it to the other primitives. I'm not even sure if this is the right wording.

Again,  all in all maybe we should merge the mechanism and migrate existing docs first, and add blurbs later? The mechanism is more important and I sure have a lot to bikeshed about the blurbs.

Reply:
Applied